### PR TITLE
Convert MarkerDetectorSpatialLocalizer to a base class

### PR DESCRIPTION
**Pull request by chrisfromwork**
We need to different localizer ids for MarkerDetectorSpatialLocalizers that support different markers. This is achieved by converting our original MarkerDetectorSpatialLocalizers to a base class. We know have ArUcoMarkerDetectorSpatialLocalizer and QRCodeMarkerDetectorSpatialLocalizer as unique localizers.

### CODE REVIEW

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment.meta [hunk 1]
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
-guid: 48efe97ab9de41547a64fd2b63d9c457
+guid: 2a03537a421f0144489f130406659dc5
```
Comment (matthejo) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment.meta]: The mysterious meta file changes continue...

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/PhysicalMarker/QRCodeMarkerDetectorSpatialLocalizer.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.SpectatorView
+{
+    public class QRCodeMarkerDetectorSpatialLocalizer : MarkerDetectorSpatialLocalizer
```
Comment (matthejo) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/PhysicalMarker/QRCodeMarkerDetectorSpatialLocalizer.cs]: >QRCodeMarkerDetectorSpatialLocalizer [](start = 17, length = 36)

Do QR codes use the same [0-n] index-based ID system? Thinking about whether or not different types of marker detectors might warrant different settings.

Comment (chrisfromwork) [reply src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/PhysicalMarker/QRCodeMarkerDetectorSpatialLocalizer.cs]: yeah they still use that as well

Comment (chrisfromwork) [reply src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/PhysicalMarker/QRCodeMarkerDetectorSpatialLocalizer.cs]: qr codes used are a "sv" prefix followed by an index. Right now we only support up to index 19.

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment.meta (hunk 1 above)
Comment (chrisfromwork) [reply src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment.meta]: i'm going to revert this.

File: src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/CompositorWindowBase.cs [hunk 3]
```diff
@@ -1,358 +1,358 @@
...
+                }
+
+                EditorGUILayout.Space();
+
+                if (GUILayout.Button(new GUIContent("Locate Shared Spatial Coordinate", "Detects the shared location used to position objects in the same physical location on multiple devices")))
+                {
+                    CompositorWorldAnchorLocalizationManager.Instance.RunRemoteLocalizationWithWorldAnchorPersistence(spatialCoordinateSystemParticipant, ArUcoMarkerDetectorSpatialLocalizer.Id, new MarkerDetectorLocalizationSettings
+                    {
+                        MarkerID = 0,
+                        MarkerSize = 0.067f,
```
Comment (matthejo) [inline src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/CompositorWindowBase.cs]: >MarkerSize = 0.067f, [](start = 24, length = 20)

Can we leave this as 0.1f for now until I've finished adding the UI for choosing this value?

**Pull request merged**